### DISCUSSION TITLE
Using GPSLogger on Android

### DISCUSSION THREAD

**Discussion by werner-kapferer-lgs** [Freika/dawarich#118] [Show and tell]
Hi all!

The README gives an example how to use OverLand and OwnTracks as apps for Android.
I now had the problem that OwnTracks was behaving very weird (I don't remember the details right now but I was not satisfied) and OverLand was not accurate and the app itself was not updated since 3 years.

Instead, I switched to [GPSLogger](https://gpslogger.app/) now!
GPSLogger provides the possibility to log to custom URLs which is working by using the OwnTracks [JSON](https://owntracks.org/booklet/tech/json/)/[HTTP](https://owntracks.org/booklet/tech/http/) specification and API from dawarich.

The configuration of GPSLogger is pretty simple and straightforward, so I want to share it with you:

- URL: `http://<your-dawarich-instance>/api/v1/owntracks/points?api_key=YOUR_API_KEY`
- HTTP Body:
```json
{
    "_type" : "location",
    "t": "u",
    "acc": "%ACC",
    "alt": "%ALT",
    "batt": "%BATT",
    "bs": "%ISCHARGING",
    "lat": "%LAT",
    "lon": "%LON",
    "tst": "%TIMESTAMP",
    "vel": "%SPD_KPH"
}
```
- HTTP Headers:
```
Content-Type: application/json
```
- HTTP Method: `POST`

I hope this helps follow dawarich fans using Android devices :wink:.
Thanks @Freika for the amazing platform!

**Comment by Freika** [comment]
Really appreciate it! Added your guide to the wiki! https://github.com/Freika/dawarich/wiki/How-to-track-your-location-to-Dawarich-with-GPSLogger

**Comment by piotr-bujnowski** [comment]
I noticed that normal json format doesn't work now in GPSLogger, maybe they changed it later.
This format works.
```
_type=location&t=u&acc=%ACC&alt=%ALT&batt=%BATT&bs=%ISCHARGING&lat=%LAT&lon=%LON&tst=%TIMESTAMP&vel=%SPD_KPH
```
Maybe I was doing something wrong with pasting json but only this worked for me.

**Reply by werner-kapferer-lgs** [reply]
Hmm. It's still working for me as described.
GPS Logger Version 132.

**Reply by AndVos** [reply]
> I noticed that normal json format doesn't work now in GPSLogger, maybe they changed it later. This format works.
> 
> ```
> _type=location&t=u&acc=%ACC&alt=%ALT&batt=%BATT&bs=%ISCHARGING&lat=%LAT&lon=%LON&tst=%TIMESTAMP&vel=%SPD_KPH
> ```
> 
> Maybe I was doing something wrong with pasting json but only this worked for me.

Hello,
I can confirm this. With GPSLogger 133 and DaWarIch 0.23.5 only then quoted version works. The Body mentioned earlier does not work anymore. Tried it today.

Grettings,
Andreas

**Reply by woernsn** [reply]
Interesting.. For me, the originally mentioned method still works with GPSLogger 133 and DaWarIch 0.23.5.

**Reply by klm42** [reply]
Same here, just set up GPSLogger 133 with Dawarich 0.23.5 today, works fine.

Did you maybe forget to set the content-type?? (wild guess)

**Reply by AndVos** [reply]
Strange. Today I recognized that data was not transferred to dawarich anymore. Changed the dataformat back to the format mentioned in the inital post and now this works. I am confused.....

**Comment by xep22** [comment]
I'm interested on you settings of the app. For me the app is logging far too rarely, every 1-2 minute, although I have set a low interval. It only logs every 5 seconds at the value of 0.

**Reply by woernsn** [reply]
My settings are:
* Log GPS/GNSS locations: on
* Log network locations: on
* Logging interval: 10s
* Distance filter: 10m
* Accuracy Filter: 60m
* Duration to match accuracy: 60s
* Absolute time to GPS fix: 120s

**Comment by MrColumbo** [comment]
Just wondering if somebody is rather using gpslogger to upload a daily gpx file to the watched folder. It looks like that GPS logger is not retrying to send a point when it failed. If you are hiking somewhere with no data coverage then there will be a gap.

**Reply by OneFrenchLeft** [reply]
I did this manualy last time, had gpslogger do a gpx and uploaded it in the evening

**Reply by Handkep** [reply]
It would be nice if it could cache all not delivered position data and send it if the connection comes back.

Edit:
I found this in the documentation:

If your phone goes offline, then the app will queue these requests until a data connection becomes available. This behavior can be changed with the 'Discard offline locations' toggle.

**Reply by werner-kapferer-lgs** [reply]
I can confirm that this is working.
In my case I use the "Send on Wi-Fi only" in addition.

**Comment by shrippen** [comment]
Thanks for the tutorial on how to setup GPSLogger. It works quite well. The only problem that now comes up is the current speed. The shown current speed in the map is way too high. Like 90 km/h when driving with my bike through town. The difference I can spot is that in comparison to my previous tracking app owntrack the current speed has way more detail after the dot. Owntrack reported for example 20.34 km/h where GPSLogger would show 20.34235453343

Could that be the reason and if, how to fix this?

**Reply by Freika** [reply]
Could this be your case? https://github.com/Freika/dawarich/releases/tag/0.24.0

**Reply by shrippen** [reply]
Ah, i found my mistake. I took the settings from https://github.com/Freika/dawarich/discussions/118#discussion-6923665 not from the documentation. In the docs SPD_KPH has been changed to SPD.

**Reply by ovizii** [reply]
> Ah, i found my mistake. I took the settings from [#118 (comment)](https://github.com/Freika/dawarich/discussions/118#discussion-6923665) not from the documentation. In the docs SPD_KPH has been changed to SPD.

Well spotted, I made the same mistake.

**Comment by Dixet** [comment]
I just started using GPS Logger as a secondary source. I noticed the tracker_id in Dawarich remained empty with this setup, so I added the "tid" parameter, which works!

Full json is now:
```
{
    "_type": "location",
    "tid": "gpslogger",
    "t": "u",
    "acc": "%ACC",
    "alt": "%ALT",
    "batt": "%BATT",
    "bs": "%ISCHARGING",
    "lat": "%LAT",
    "lon": "%LON",
    "tst": "%TIMESTAMP",
    "vel": "%SPD"
}
```
